### PR TITLE
Bah 2164 deploy test reports in GitHub pages and enhance slack notification

### PR DESCRIPTION
**Pull request by gokultw**
- Deploy HTML reports
- Delete reports more than 10
- Custom 404 page to handle scenarios invalid URLs, handle 404 errors for valid URL  & handle Report Expiry scenarios.
- Modified Slack notification with HTML report details.

image

image

Deploy test reports in Github pages and Enhance slack notification
https://bahmni.atlassian.net/browse/BAH-2164

### CODE REVIEW

File: .github/workflows/repository_dispatch.yml [hunk 1]
```diff
@@ -66,18 +66,53 @@ jobs:
...
+            then  
+              echo "deleting reports older than $MAX_DIR occurrences"
+              ls -d report-* | head -$deleteDir | xargs rm -rf
+            fi
+            count=$(ls -d report-* | wc -l)
+            count=`echo $count | sed -e 's/^[[:space:]]*//'`
+            echo total files after deletion $count
+            git add .
+            git commit -m "Deploy Reports"
+            git push origin $TARGET_BRANCH
```
Comment (N0-man) [inline .github/workflows/repository_dispatch.yml:101]: you are duplicating the code between dispatch and main... atleast extract this into its own script to reuse.

Comment (gokultw) [reply .github/workflows/repository_dispatch.yml]: reused the same code for both the workflows using a bash script.

**Review by N0-man — APPROVED** [review]
Slack notification can be extracted into its own action instead of duplicating

**Comment by gokultw** [comment]
> Slack notification can be extracted into its own action instead of duplicating

And call trigger than action using repository dispatch?

**Pull request merged**